src/Symfony/Component/Routing/RouteCompiler.php

```diff
@@ -184,6 +186,12 @@ private static function compilePattern(Route $route, string $pattern, bool $isHo
                 $regexp = self::transformCapturingGroupsToNonCapturings($regexp);
             }
 
+            // Fixed variable, could be transformed as text token
+            if ($fixStaticVariables && preg_quote($regexp, self::REGEX_DELIMITER) === $regexp) {
```

Comment:
does this really work?
if the static data is `a.b`, $regexp will be `a\.b`:
- `preg_quote($regexp, self::REGEX_DELIMITER)` => `a\\\.b` !== `$regexp`

Instead, we should do `preg_quote(unpreg_quote($regexp), self::REGEX_DELIMITER) === $regexp` - with `unpreg_quote()` to be written.